src/include/ZividPython/Wrappers.h

```diff
@@ -13,6 +16,50 @@
 
 namespace ZividPython
 {
+    namespace
+    {
+        std::string toSnakeCase(const std::string capitalized)
+        {
+            if(capitalized.empty())
+            {
+                throw std::invalid_argument{ "String is empty." };
+            }
+
+            if(!isupper(capitalized[0]))
+            {
+                throw std::runtime_error{ "First character of string: '" + capitalized + "' is not capitalized" };
+            }
+            std::stringstream ss;
+            ss << char(tolower(capitalized[0]));
+
+            for(auto i = 1; i < capitalized.size(); ++i)
+            {
+                if(isupper(capitalized[i]))
+                {
+                    auto previous = i - 1;
+                    auto next = i + 1;
+
+                    // if surrounded by capital case (looking at the B character): ABC -> abc
```

Comment:
This comment is incomplete, since the following if-statement also introduces special handling if the final letter is capitalized, regardless of whether it is surrounded by capital case.

The current code yields:
`CamelCaseA` -> `camel_casea`
Is that intended behavior?

Reply:
No. or condition was in the wrong scope, when I moved it to the correct location it works in that test case.

Reply:
Tested new version. Looks good.